src/registrar/models/domain_application.py

```diff
@@ -707,12 +727,16 @@ def reject(self):
         As side effects this will delete the domain and domain_information
         (will cascade), and send an email notification."""
         if self.status == self.ApplicationStatus.APPROVED:
-            domain_state = self.approved_domain.state
-            # Only reject if it exists on EPP
-            if domain_state != Domain.State.UNKNOWN:
-                self.approved_domain.deletedInEpp()
-            self.approved_domain.delete()
-            self.approved_domain = None
+            try:
```

Comment:
This should never fail in the real world but running tests made me realize that it can be solidified a bit

Reply:
Glad you added this, this was particularly annoying when running scripts